Startup crash when signed into FxA

## Steps to reproduce
- Start up app
- Sign into FxA

### Expected behavior
No crash.

### Actual behavior
Crash. Note: the crash also occurs as soon as the user successfully signs into FxA.

It looks like we're calling View code from not the main thread with RxJava:
```
8-28 19:42:36.996 28039-28273/org.mozilla.tv.firefox.debug E/AndroidRuntime: FATAL EXCEPTION: pool-6-thread-1
    Process: org.mozilla.tv.firefox.debug, PID: 28039
    io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalStateException: Method call should happen from the main thread.
        at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
        at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
        at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
        at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:67)
        at io.reactivex.internal.operators.observable.ObservableHide$HideDisposable.onNext(ObservableHide.java:67)
        at io.reactivex.internal.util.NotificationLite.accept(NotificationLite.java:246)
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.test(BehaviorSubject.java:569)
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.emitNext(BehaviorSubject.java:564)
        at io.reactivex.subjects.BehaviorSubject.onNext(BehaviorSubject.java:268)
        at org.mozilla.tv.firefox.fxa.FxaRepo$FirefoxAccountObserver.onProfileUpdated(FxaRepo.kt:141)
        at mozilla.components.service.fxa.manager.FxaAccountManager$stateActions$5.invoke(FxaAccountManager.kt:649)
        at mozilla.components.service.fxa.manager.FxaAccountManager$stateActions$5.invoke(FxaAccountManager.kt:101)
        at mozilla.components.support.base.observer.ObserverRegistry.notifyObservers(ObserverRegistry.kt:131)
        at mozilla.components.service.fxa.manager.FxaAccountManager.notifyObservers(FxaAccountManager.kt)
        at mozilla.components.service.fxa.manager.FxaAccountManager.stateActions(FxaAccountManager.kt:648)
        at mozilla.components.service.fxa.manager.FxaAccountManager$processQueueAsync$1.invokeSuspend(FxaAccountManager.kt:451)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: java.lang.IllegalStateException: Method call should happen from the main thread.
        at com.squareup.picasso.Utils.checkMain(Utils.java:127)
        at com.squareup.picasso.RequestCreator.into(RequestCreator.java:679)
        at com.squareup.picasso.RequestCreator.into(RequestCreator.java:665)
        at org.mozilla.tv.firefox.channels.ImageSetStrategy$ByPath.invoke(ChannelTile.kt:104)
        at org.mozilla.tv.firefox.navigationoverlay.NavigationOverlayFragment$observeAccountState$1.accept(NavigationOverlayFragment.kt:265)
        at org.mozilla.tv.firefox.navigationoverlay.NavigationOverlayFragment$observeAccountState$1.accept(NavigationOverlayFragment.kt:87)
        at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
        at io.reactivex.internal.operators.observable.ObservableHide$HideDisposable.onNext(ObservableHide.java:67) 
        at io.reactivex.internal.util.NotificationLite.accept(NotificationLite.java:246) 
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.test(BehaviorSubject.java:569) 
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.emitNext(BehaviorSubject.java:564) 
        at io.reactivex.subjects.BehaviorSubject.onNext(BehaviorSubject.java:268) 
        at org.mozilla.tv.firefox.fxa.FxaRepo$FirefoxAccountObserver.onProfileUpdated(FxaRepo.kt:141) 
        at mozilla.components.service.fxa.manager.FxaAccountManager$stateActions$5.invoke(FxaAccountManager.kt:649) 
        at mozilla.components.service.fxa.manager.FxaAccountManager$stateActions$5.invoke(FxaAccountManager.kt:101) 
        at mozilla.components.support.base.observer.ObserverRegistry.notifyObservers(ObserverRegistry.kt:131) 
        at mozilla.components.service.fxa.manager.FxaAccountManager.notifyObservers(FxaAccountManager.kt) 
        at mozilla.components.service.fxa.manager.FxaAccountManager.stateActions(FxaAccountManager.kt:648) 
        at mozilla.components.service.fxa.manager.FxaAccountManager$processQueueAsync$1.invokeSuspend(FxaAccountManager.kt:451)
```

### Device information
* Fire TV device: ?
* Latest affected Firefox version: v4.4-LAT1
* Earliest affected Firefox version: ?
* Turbo Mode: ?

--- a/app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt
+++ b/app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt
@@ -25,6 +25,7 @@ import androidx.core.widget.NestedScrollView
 import androidx.fragment.app.Fragment
 import androidx.transition.Fade
 import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.disposables.Disposable
 import io.reactivex.rxkotlin.addTo
@@ -259,24 +260,26 @@ class NavigationOverlayFragment : Fragment() {
     private fun observeAccountState(): Disposable {
         val fxaRepo = serviceLocator.fxaRepo
 
-        return fxaRepo.accountState.subscribe { accountState ->
-            when (accountState) {
-                is AccountState.AuthenticatedWithProfile -> {
-                    accountState.profile.avatarSetStrategy.invoke(fxaButton)
-                    fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
-                }
-                AccountState.AuthenticatedNoProfile -> {
-                    fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
-                    fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
-                }
-                AccountState.NeedsReauthentication, AccountState.NotAuthenticated -> {
-                    fxaButton.setImageResource(R.drawable.ic_fxa_login)
-                    fxaButton.contentDescription =
+        return fxaRepo.accountState
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { accountState ->
+                when (accountState) {
+                    is AccountState.AuthenticatedWithProfile -> {
+                        accountState.profile.avatarSetStrategy.invoke(fxaButton)
+                        fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
+                    }
+                    AccountState.AuthenticatedNoProfile -> {
+                        fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
+                        fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
+                    }
+                    AccountState.NeedsReauthentication, AccountState.NotAuthenticated -> {
+                        fxaButton.setImageResource(R.drawable.ic_fxa_login)
+                        fxaButton.contentDescription =
                             resources.getString(R.string.fxa_navigation_item_new,
-                                    resources.getString(R.string.app_name))
+                                resources.getString(R.string.app_name))
+                    }
                 }
             }
-        }
     }
 
     private fun observePinnedTiles(): Disposable {

